.openhands/microagents/glossary.md

```diff
@@ -0,0 +1,175 @@
+# OpenHands Glossary
+
+### Agent
+The core AI entity in OpenHands that can perform software development tasks by interacting with tools, browsing the web, and modifying code.
+
+#### Agent Controller
+A component that manages the agent's lifecycle, handles its state, and coordinates interactions between the agent and various tools.
+
+#### Agent Delegation
+The ability of an agent to hand off specific tasks to other specialized agents for better task completion.
+
+#### Agent Hub
+A central registry of different agent types and their capabilities, allowing for easy agent selection and instantiation.
+
+#### Agent Skill
+A specific capability or function that an agent can perform, such as file manipulation, web browsing, or code editing.
+
+#### Agent State
+The current context and status of an agent, including its memory, active tools, and ongoing tasks.
+
+#### CodeAct Agent
+A specialized agent type in OpenHands designed specifically for software development tasks and code manipulation.
+
+### Browser
+A system for web-based interactions and tasks.
+
+#### Browser Environment
+
+#### Browser Gym
+A testing and evaluation environment for browser-based agent interactions and tasks.
+
+#### Web Browser Tool
+A tool that enables agents to interact with web pages and perform web-based tasks.
+
+### Commands
+Terminal and execution related functionality.
+
+#### Bash Parser
+A component that processes and interprets bash commands and their outputs for agent interaction.
```

Comment:
I think this implies an interaction between Bash Parser, Bash Session, and the Runtime, but it isn't spelled out. Also Bash is one specific shell, we might consider calling this Shell or Console at some point.

Reply:
Now we have tmux integrated, i don't think we have a explicit bash parser anymore, i.e., most things are handled by tmux automatically. Maybe we could remove this to reduce confusion?